### PR TITLE
Test more windows targets

### PR DESCRIPTION
**Pull request by justsmth**
### Description of changes: 
* Add CI testing for 'i686-pc-windows-msvc' and 'i686-pc-windows-gnu'
* For Windows, set `static_crt` for our build. See: #453 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and the ISC license.

### CODE REVIEW

File: .github/workflows/cross.yml [hunk 1]
```diff
@@ -142,6 +144,8 @@ jobs:
     if: github.repository_owner == 'aws'
     name: ${{ matrix.target }}
     runs-on: windows-latest
+    env:
+      RUSTFLAGS: '-Clink-arg=/WX'
```
Comment (rami3l) [inline .github/workflows/cross.yml:148]: `/WX` is a `link.exe`-exclusive thing, I believe.

File: .github/workflows/cross.yml [hunk 2]
```diff
@@ -151,16 +172,20 @@ jobs:
...
+      - name: Release test on `${{ matrix.target }}`
+        shell: bash
+        run: cargo test -p aws-lc-rs --release --target ${{ matrix.target }} --features bindgen
 
   aws-lc-rs-windows-arm64:
     if: github.repository_owner == 'aws'
     name: aarch64-pc-windows-msvc
     runs-on: windows-latest
+    env:
+      RUSTFLAGS: '-Clink-arg= /WX'
```
Comment (rami3l) [inline .github/workflows/cross.yml:188]: So it looks like `RUSTFLAGS: -Clink-arg=/WX` works (without the space), but not on bash.

File: .github/workflows/cross.yml [hunk 3]
```diff
@@ -134,14 +134,23 @@ jobs:
...
+  aws-lc-rs-windows-cross:
     if: github.repository_owner == 'aws'
-    name: x86_64-pc-windows-gnu
+    name: ${{ matrix.target }}
     runs-on: windows-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        target:
+          - 'x86_64-pc-windows-gnu'
```
Comment (rami3l) [inline .github/workflows/cross.yml:149]: Still, it makes no sense to `-Clink-arg=/WX` if we're using the GNU toolchain...

Comment (justsmth) [reply .github/workflows/cross.yml:149]: Yeah, I've done some refactoring in the latest commit to separate the Windows CI jobs building for "gnu" versus "msvc".

**Comment by justsmth** [comment]
It appears that the symbols provided by the assembly code are not getting prefixed:
```
% dumpbin //EXPORTS //SYMBOLS  ./target/aarch64-pc-windows-msvc/debug/build/aws-lc-sys-bdeeea1c963418e6/out/build/artifacts/aws_lc_0_19_0_crypto.lib | grep External | grep SECT1 | egrep -v aws_lc | grep 'bn_'
006 00000000 SECT1  notype       External     | bn_add_words
00A 00000050 SECT1  notype       External     | bn_sub_words
006 00000000 SECT1  notype ()    External     | bn_mul_mont
```

```
% dumpbin //EXPORTS //SYMBOLS  ./target/aarch64-pc-windows-msvc/debug/build/aws-lc-sys-bdeeea1c963418e6/out/build/artifacts/aws_lc_0_19_0_crypto.lib | grep External | grep SECT1 | grep 'aws_lc' | head -n 5
047 00000000 SECT13 notype ()    External     | aws_lc_0_19_0_aes_nohw_ctr32_encrypt_blocks
04C 00000000 SECT14 notype ()    External     | aws_lc_0_19_0_aes_nohw_cbc_encrypt
051 00000000 SECT15 notype ()    External     | aws_lc_0_19_0_AES_wrap_key
056 00000000 SECT16 notype ()    External     | aws_lc_0_19_0_AES_unwrap_key
05B 00000000 SECT17 notype ()    External     | aws_lc_0_19_0_AES_wrap_key_padded
```

**Comment by justsmth** [comment]
> It appears that the symbols provided by the assembly code are not getting prefixed:

I was missing the required include directory when building the assembly files. I think I'm closing in on the solution now.

**Review by skmcgrail — APPROVED** [review]

File: .github/workflows/cross.yml [hunk 4]
```diff
@@ -132,45 +132,80 @@ jobs:
...
+        run: |
+          if ('${{ matrix.target }}' -like '*aarch64*') {
+            echo "ACTION_CARGO=build" | Out-File -FilePath $env:GITHUB_ENV -Append
+          } else {
+            echo "ACTION_CARGO=test" | Out-File -FilePath $env:GITHUB_ENV -Append
+          }
+      - name: Set RUSTFLAGS
+        run: |
+          if ('${{ matrix.crt_static }}' -eq '1') {
+            echo "RUSTFLAGS=-Clink-arg=/WX -Ctarget-feature=+crt-static" | Out-File -FilePath $env:GITHUB_ENV -Append
```
Comment (samuel40791765) [inline .github/workflows/cross.yml:204]: I might have missed this, but what does `-Clink-arg=/WX` actually do?

**Review by samuel40791765 — APPROVED** [review]

File: .github/workflows/cross.yml (hunk 4 above)
Comment (justsmth) [reply .github/workflows/cross.yml:204]: The "[/WX](https://learn.microsoft.com/en-us/cpp/build/reference/wx-treat-linker-warnings-as-errors?view=msvc-170)" force linker warning to surface as errors in the build.  It helps us avoid having "false negatives" where our Windows build succeeds even though there are problems with the linkage.

**Pull request merged**